src/Hedgehog/Shrink.fs

```diff
@@ -1,42 +1,38 @@
 ﻿namespace Hedgehog
 
-open FSharpx.Collections
-
-module LazyList =
-    let singleton (x : 'a) : LazyList<'a> =
-        LazyList.cons x LazyList.empty
-
-    let concatMap (f : 'a -> LazyList<'b>) (xs : LazyList<'a>) : LazyList<'b> =
-        LazyList.concat <|
-        LazyList.map f xs
-
-    let consNub (x : 'a) (ys0 : LazyList<'a>) : LazyList<'a> =
-        match ys0 with
-        | LazyList.Nil ->
-            singleton x
-        | LazyList.Cons (y, ys) ->
+module SeqExtra =
+    let cons (x : 'a) (xs : seq<'a>) : seq<'a> = 
+        seq {
+            yield x
+            yield! xs
+        }
+
+    let consNub (x : 'a) (ys0 : seq<'a>) : seq<'a> =
+        match Seq.tryHead ys0 with
+        | None -> Seq.singleton x
+        | Some y ->
             if x = y then
-                LazyList.cons y ys
+                ys0
             else
-                LazyList.cons x <| LazyList.cons y ys
+                cons x ys0
+
 
 module Shrink =
     /// Produce all permutations of removing 'k' elements from a list.
-    let removes (k0 : int) (xs0 : List<'a>) : LazyList<List<'a>> =
-        let rec loop (k : int) (n : int) (xs : List<'a>) : LazyList<List<'a>> =
+    let removes (k0 : int) (xs0 : List<'a>) : seq<List<'a>> =
+        let rec loop (k : int) (n : int) (xs : List<'a>) : seq<List<'a>> =
             let hd = List.take k xs
             let tl = List.skip k xs
```

Comment:
The failing test is due to the `List.take / skip` functions from the base class library. The current used `List.take / skip` function are from `FSharpx.Collections`, which don't throw if the `take`(or `skip`) number is great  than the list length.

We can address this issue as below

```fsharp
let k' = Math.Min(k, xs.length)
let hd = List.take k' xs
let hd = List.take k' xs
```

Reply:
@jwChung, thanks for looking into it! Would you care to submit a PR on that branch? Otherwise I can try it myself and of course give you credit :+1:

Reply:
@moodmosaic [I have submitted my PR](https://github.com/Porges/fsharp-hedgehog/pull/1) to @Porges's branch named bench.

Reply:
Ping @Porges 🚀

Reply:
thanks for doing this @jwChung, I haven't had any time to dedicate to this recently :+1: